### PR TITLE
Enable AD integrated tests for non-Windows

### PR DESCRIPTION
**Pull request by lilgreenbird**
Also add ADAL exception info to error message to help debug errors

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerADAL4JUtils.java [hunk 1]
```diff
@@ -41,25 +41,30 @@ static SqlFedAuthToken getSqlFedAuthToken(SqlFedAuthInfo fedAuthInfo, String use
             return new SqlFedAuthToken(authenticationResult.getAccessToken(), authenticationResult.getExpiresOnDate());
         } catch (MalformedURLException | InterruptedException e) {
             throw new SQLServerException(e.getMessage(), e);
-        } catch (ExecutionException e) {
+        } catch (Exception e) {
```
Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerADAL4JUtils.java:44]: Blanket catching all exceptions is supposed to be a code smells. Is there a way for us to not do this?

File: src/main/java/com/microsoft/sqlserver/jdbc/AuthenticationJNI.java [hunk 2]
```diff
@@ -153,7 +153,7 @@ private static native int SNISecGenClientContext(byte[] psec, int[] secptrsize,
 
     private static native int GetDNSName(String address, String[] DNSName, java.util.logging.Logger log);
 
-    private static native FedAuthDllInfo ADALGetAccessTokenForWindowsIntegrated(String stsURL,
+    private static synchronized native FedAuthDllInfo ADALGetAccessTokenForWindowsIntegrated(String stsURL,
```
Comment (ulvii) [inline src/main/java/com/microsoft/sqlserver/jdbc/AuthenticationJNI.java:156]: Please remove this.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerADAL4JUtils.java [hunk 3]
```diff
@@ -41,25 +41,34 @@ static SqlFedAuthToken getSqlFedAuthToken(SqlFedAuthInfo fedAuthInfo, String use
             return new SqlFedAuthToken(authenticationResult.getAccessToken(), authenticationResult.getExpiresOnDate());
         } catch (MalformedURLException | InterruptedException e) {
             throw new SQLServerException(e.getMessage(), e);
-        } catch (ExecutionException e) {
+        } catch (ExecutionException | AuthenticationException e) {
+            if (adal4jLogger.isLoggable(Level.FINE)) {
```
Comment (ulvii) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerADAL4JUtils.java:45]: Please change this to SEVERE level.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerADAL4JUtils.java [hunk 4]
```diff
@@ -90,7 +99,11 @@ static SqlFedAuthToken getSqlFedAuthTokenIntegrated(SqlFedAuthInfo fedAuthInfo,
             return new SqlFedAuthToken(authenticationResult.getAccessToken(), authenticationResult.getExpiresOnDate());
         } catch (InterruptedException | IOException e) {
             throw new SQLServerException(e.getMessage(), e);
-        } catch (ExecutionException e) {
+        } catch (ExecutionException | AuthenticationException e) {
+            if (adal4jLogger.isLoggable(Level.FINE)) {
```
Comment (ulvii) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerADAL4JUtils.java:103]: Please change this to SEVERE level.

**Review by ulvii — APPROVED** [review]

**Review by rene-ye — APPROVED** [review]

**Review by peterbae — APPROVED** [review]

**Pull request merged**